plans/cd4pe_job.pp

```diff
@@ -7,21 +7,38 @@
   Optional[String[1]]             $docker_image = undef,
   Optional[Array[String[1]]]      $docker_run_args = undef,
   Optional[String[1]]             $docker_pull_creds = undef,
+  Optional[String[1]]             $base_64_registry_ca_cert = undef,
   Optional[String[1]]             $base_64_ca_cert = undef,
 ) {
 
   $cd4pe_token = system::env('CD4PE_TOKEN')
 
-  return run_task(
-    'cd4pe_jobs::run_cd4pe_job',
-    $targets,
+  $basic_args = {
```

Comment:
This pattern to build a hash is annoying. One thing that might help is to prefix with `_` when they're meant to be intermediate variables. Could help avoid failing to pass the last version to the task.